src/aap_eda/api/views/project.py

```diff
@@ -211,6 +221,27 @@ def partial_update(self, request, pk):
 
         update_fields = []
         for key, value in serializer.validated_data.items():
+            if key == "proxy":
+                if value and not (
+                    value.startswith("http://") or value.startswith("https://")
+                ):
+                    value = f"http://{value}"
+                if ENCRYPTED_STRING in value:
+                    if (
+                        project.proxy
+                        and get_proxy_for_display(
+                            project.proxy.get_secret_value()
+                        )
+                        == value
+                    ):
+                        continue  # value is unchanged. skip
+
+                    error = (
+                        "The password in the proxy field should be unencrypted"
+                    )
+                    return Response(
+                        {"errors": error}, status=status.HTTP_400_BAD_REQUEST
+                    )
```

Comment:
can you give more context about this logic? Why a partial update would receive a proxy with the content encrypted by the application?

Reply:
I see your point. Would it not be better to run the transformation against the request.data so you can know in advance if the field is updated or not?